pyproject.toml

```diff
@@ -58,7 +58,7 @@ test = [
     "blacken-docs==1.9.2",
     "black~=22.0",
     "coverage[toml]",
-    "fsspec<2023.9", # Temporary, newer version causing "test_no_versions_with_cloud_protocol" to fail
+    "fsspec>=2021.4",
```

Comment:
Is the new error handling compatible with `2021.4<=version<2023.9`?

Reply:
In the older versions the error will be caught later on in this check https://github.com/kedro-org/kedro/blob/main/kedro/io/core.py#L544 

It's not possible to keep it fully backwards compatible, because the test-coverage would then fail and in different ways depending on which version of `fsspec` is installed. It's just a test-requirement though, so backward compatibility isn't as important as for the regular dependencies.

Reply:
Could we then pin the test requirement at `>2023.9`?

Reply:
I ended up just removing `fsspec` from the test requirements, because it wasn't included before the pin was needed.